common/persistence/cassandra/cassandraPersistence.go

```diff
@@ -2718,8 +2718,8 @@ func (d *cassandraPersistence) CreateTasks(
 				rowTypeTask,
 				task.TaskID,
 				domainID,
-				task.Execution.GetWorkflowId(),
-				task.Execution.GetRunId(),
+				*task.Execution.WorkflowId,
```

Comment:
I agree with your concern about the type. One way is to have a unit test for it. I think another way is to have cast the type first before passing it in.

var workflowID string
workflowID = *task.Execution.WorkflowId